### PR TITLE
/bookshelf/remove endpoint complete

### PR DESCRIPTION
**Pull request by ElliotSchumacher**
Added support for deleting bookshelf records for specified users. I've tested 8 edge cases and pass each one.

### CODE REVIEW

**Review by ASimpleFox — CHANGES_REQUESTED** [review]
If testing is complete, recommend that we get rid of the console.log() statements in here. Also I noticed that you have this comment that says "I'll swap out this if statement with the method Nicholas wrote", should we get that swapped in before pushing?

File: api/server.js [hunk 1]
```diff
@@ -171,11 +171,34 @@ app.post("/bookshelves/add", async (req, res) => {
...
+    try {
+        res.type("JSON");
+        let username = req.body.username;
+        let bookshelf = req.body.bookshelf;
+        let isbn = req.body.isbn;
+        if (!username || !bookshelf || !isbn) {
+            res.status(CLIENT_ERROR_CODE_400).send("Missing one or more required body parameters");
+        } else  {
+            let userId = await getUserId(username);
+            console.log("userId: " + userId);
```
Comment (ASimpleFox) [inline api/server.js:183]: delet

File: api/server.js [hunk 2]
```diff
@@ -316,6 +339,36 @@ async function checkIfExist(username) {
...
 
+/**
+ * Returns the id of the user with the given username or 0 if no user exists.
+ * @param {String} username - The username of the user to get the id for.
+ * @returns {int} The users id or 0 if no user exists with the username.
+ */
+async function getUserId(username) {
+    let query = "SELECT id FROM User WHERE User.username = ?;"
+    let [rows] = await db.query(query, [username]);
+    console.log("rows: " + rows);
```
Comment (ASimpleFox) [inline api/server.js:350]: delet

File: api/server.js [hunk 3]
```diff
@@ -316,6 +339,36 @@ async function checkIfExist(username) {
...
+ * Removes a given book from a specified bookshelf for a given user.
+ * @param {int} userId - The id for the given user.
+ * @param {String} bookshelf - The name of the bookshelf to alter.
+ * @param {int} isbn - The isbn of the book to remove.
+ * @return {boolean} True if the table was altered, false otherwise.
+ */
+async function deleteBookshelfRecord(userId, bookshelf, isbn) {
+    let query = "DELETE FROM Bookshelf WHERE id_user = ? AND isbn = ? AND shelf_name = ?;";
+    let [rows] = await db.query(query, [userId, isbn, bookshelf])
+    console.log(rows.affectedRows);
```
Comment (ASimpleFox) [inline api/server.js:368]: delet

**Review by ASimpleFox — APPROVED** [review]
Looks good! I'll need to pull and refactor this code after it merges with master oof

**Comment by ElliotSchumacher** [comment]
Please don't delete the branch

**Pull request merged**